src/test/java/com/google/jenkins/plugins/computeengine/ComputeEngineCloudWindowsIT.java

```diff
@@ -200,37 +218,70 @@ public void testWorkerCreated() throws Exception {
         assertEquals(logs(),3, i.getLabels().size());
 
         // Instance should have a label with key CONFIG_LABEL_KEY and value equal to the config's name prefix
-        assertEquals(logs(), validInstanceConfiguration1().namePrefix, i.getLabels().get(ComputeEngineCloud.CONFIG_LABEL_KEY));
+        assertEquals(logs(), ic.namePrefix, i.getLabels().get(ComputeEngineCloud.CONFIG_LABEL_KEY));
         // proper id label to properly count instances
         assertEquals(logs(), cloud.getInstanceUniqueId(), i.getLabels().get(ComputeEngineCloud.CLOUD_ID_LABEL_KEY));
     }
 
-    private static InstanceConfiguration validInstanceConfiguration1() throws Exception{
-        GoogleKeyPair kp = GoogleKeyPair.generate("");
-        // Have to reformat since GoogleKeyPair's format is for metadata server and not typical public key format
-        String publicKey = kp.getPublicKey().trim().substring(1);
+    // Tests snapshot is created when we have failure builds for given node
+    // Snapshot creation is longer for windows vm's.
+    @Test(timeout = 0)
+    public void testSnapshotCreated() throws Exception {
+        logOutput.reset();
 
-        String startupScript = String.format("Stop-Service sshd\n" +
-                "$username = " + "\"%1$s\"\n" +
-                "$ConfiguredPublicKey = "  + "\"%2$s\"\n" +
+        ComputeEngineCloud cloud = (ComputeEngineCloud) r.jenkins.clouds.get(0);
+        cloud.addConfiguration(snapshotInstanceConfiguration());
+
+        FreeStyleProject project = r.createFreeStyleProject();
+        Builder step = new Shell("exit 1");
+        project.getBuildersList().add(step);
+        project.setAssignedLabel(new LabelAtom(SNAPSHOT_LABEL));
+
+        FreeStyleBuild build = r.assertBuildStatus(Result.FAILURE, project.scheduleBuild2(0));
+        Node worker = build.getBuiltOn();
+        try {
+            r.jenkins.getNode(worker.getNodeName()).toComputer().doDoDelete();
+
+            Snapshot createdSnapshot = client.getSnapshot(projectId, worker.getNodeName());
+            assertEquals(logs(), createdSnapshot.getStatus(), "READY");
```

Comment:
The assertNotNull should come before the assertEquals to prevent a NPE.

Reply:
done